tony-core/src/main/java/com/linkedin/tony/ApplicationMaster.java

```diff
@@ -1053,10 +1054,16 @@ public void onContainersAllocated(List<Container> containers) {
       for (Container container : containers) {
         // Need to explicitly remove container requests from remoteRequestsTable in AMRMClient, otherwise
         // resources get double-requested (YARN-1902)
+        int numGPU = 0;
+        try {
+          numGPU = (int) container.getResource().getResourceInformation(Constants.GPU_URI).getValue();
+        } catch (ResourceNotFoundException e) {
+          LOG.debug("No GPU resource configured for the cluster. ", e);
```

Comment:
Is there a better way to check for no GPU resource in cluster? Throwing and caching exception seems extreme. 
See Item 69 from Josh Bloch's Effective Java: [Use exceptions only for exceptional conditions](https://ahdak.github.io/blog/effective-java-part-9/)